sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubsScalerProvider.cs

```diff
@@ -0,0 +1,103 @@
+        {
+            AzureComponentFactory azureComponentFactory;
+            if ((triggerMetadata.Properties != null) && (triggerMetadata.Properties.TryGetValue(nameof(AzureComponentFactory), out object value)))
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+
+            var configuration = serviceProvider.GetService<IConfiguration>();
+            var hostComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            var logForwarder = serviceProvider.GetService<AzureEventSourceLogForwarder>();
+            var options = serviceProvider.GetService<IOptions<EventHubOptions>>();
+            var loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            var checkpointClientProvider = serviceProvider.GetService<CheckpointClientProvider>();
+            var nameResolver = serviceProvider.GetService<INameResolver>();
+            var evetnHubMetadata = JsonConvert.DeserializeObject<EventHubMetadata>(triggerMetadata.Metadata.ToString());
+            var factory = new EventHubClientFactory(configuration, hostComponentFactory, options, nameResolver, logForwarder, checkpointClientProvider);
+            evetnHubMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            var eventHubConsumerClient = factory.GetEventHubConsumerClient(evetnHubMetadata.EventHubName, evetnHubMetadata.Connection, evetnHubMetadata.ConsumerGroup);
+            var checkpointStore = new BlobCheckpointStoreInternal(
+                factory.GetCheckpointStoreClient(),
+                triggerMetadata.FunctionName,
+                loggerFactory.CreateLogger<BlobCheckpointStoreInternal>());
+            var eventHubMerticsProvider = new EventHubMetricsProvider(
+                triggerMetadata.FunctionName,
+                eventHubConsumerClient,
+                checkpointStore,
+                loggerFactory.CreateLogger<EventHubMetricsProvider>()
+                );
+
+            _scaleMonitor = new EventHubsScaleMonitor(
+                triggerMetadata.FunctionName,
+                eventHubConsumerClient,
+                checkpointStore,
+                loggerFactory.CreateLogger<EventHubsScaleMonitor>());
+
+            _targetScaler = new EventHubsTargetScaler(
+                triggerMetadata.FunctionName,
+                eventHubConsumerClient,
+                options.Value,
+                eventHubMerticsProvider,
+                loggerFactory.CreateLogger<EventHubsScaleMonitor>());
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            return _scaleMonitor;
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class EventHubMetadata
```

Comment:
Do we intend this to be used elsewhere?   I'm wondering if we should move it to its own file or consider marking it `private`.   It doesn't seem that it requires being in the context of the sale provider to make sense.

Reply:
`EventHubMetadata` will be used only in EventHubsScalerProvider, I changes to `private`